ffmpeg/encoder.c

```diff
@@ -308,10 +313,9 @@ static int encode(AVCodecContext* encoder, AVFrame *frame, struct output_ctx* oc
     octx->res->pixels += encoder->width * encoder->height;
   }
 
-
   // We don't want to send NULL frames for HW encoding
   // because that closes the encoder: not something we want
-  if (AV_HWDEVICE_TYPE_NONE == octx->hw_type || frame) {
+  if (AV_HWDEVICE_TYPE_NONE == octx->hw_type || AVMEDIA_TYPE_AUDIO == ost->codecpar->codec_type || frame) {
```

Comment:
Here certainly do we need to check AVMEDIA_TYPE_AUDIO , I think that should accept audio NULL frame to flush the encoder.